pkg/helm/release/manager.go

```diff
@@ -226,15 +226,22 @@ func installRelease(ctx context.Context, tiller *tiller.ReleaseServer, namespace
 		// Workaround for helm/helm#3338
 		if releaseResponse.GetRelease() != nil {
 			uninstallReq := &services.UninstallReleaseRequest{
-				Name:  releaseResponse.GetRelease().GetName(),
+				Name:  name,
 				Purge: true,
 			}
 			_, uninstallErr := tiller.UninstallRelease(ctx, uninstallReq)
-			if uninstallErr != nil {
-				return nil, fmt.Errorf("failed to roll back failed installation: %s: %s", uninstallErr, err)
+
+			// In certain cases, InstallRelease will return a partial release in
+			// the response even when it doesn't record the release in its release
+			// store. In that case the rollback will fail with a not found error
+			// because there was nothing to rollback. So we check for that and
+			// skip logging the rollback error when the error was that it didn't
+			// exist.
```

Comment:
Can you explain when does this happen, what are these cases. Just an example. 

Another thing is I would rephrase the last sentence, it doesn't read very clearly to me.